src/fileActions.jsx

```diff
@@ -209,3 +216,95 @@ export const RenameItemModal = ({ path, setPath, selected }) => {
         </Modal>
     );
 };
+
+export const EditPermissionsModal = ({ selected, path }) => {
+    const Dialogs = useDialogs();
+    const getPermissions = (str) => {
+        let permissions = 0;
+        if (str.slice(0, 1) === "r") { permissions += 4 }
+        if (str.slice(1, 2) === "w") { permissions += 2 }
+        if (str.slice(2, 3) === "x") { permissions += 1 }
+        return permissions;
+    };
+    const [owner, setOwner] = useState(selected.owner);
+    const [ownerAccess, setOwnerAccess] = useState(getPermissions(selected.permissions.slice(1, 4)));
+    const [group, setGroup] = useState(selected.group);
+    const [groupAccess, setGroupAccess] = useState(getPermissions(selected.permissions.slice(4, 7)));
+    const [otherAccess, setOtherAccess] = useState(getPermissions(selected.permissions.slice(7, 10)));
+    const [errorMessage, setErrorMessage] = useState(undefined);
+
+    const editPermissions = () => {
+        const options = { err: "message", superuser: "try" };
+        const command = ["chmod", ownerAccess + groupAccess + otherAccess, "/" + path.join("/") + "/" + selected.name];
+
+        cockpit.spawn(command, options)
```

Comment:
You should rather 'chain' the two spawn calls, with normal promise chaining.

```js
cockpit.spawn(command, options)
                .then(() =>
                      cockpit.spawn(["chown", owner + ":" + group, "/" + path.join("/") + "/" + selected.name], options)
                 )
                 .then(Dialogs.close, err => { setErrorMessage(err.message) });
```

Reply:
How abotu that you don't nest these at all actually? So unpack all nested and chain them? I think it looks much more readable.

Reply:
```javascript
cockpit.spawn(command, options)
               .then(() => cockpit.spawn(["chown", owner + ":" + group, "/" + path.join("/") + "/" + selected.name], options))
               .then(() => cockpit.spawn(renameCommand, options))
               .then(() => {
                    if (selected.items_cnt)
                        setPath(path.slice(0, -1).concat(name));
                    });
                    Dialogs.close();
               }, err => setErrorMessage(err.message) );
```